samples/snippets/src/main/java/events/ImportUserEventsBigQuery.java

```diff
@@ -0,0 +1,189 @@
+   */
+  private static final String DATASET_ID = "user_events";
+
+  /**
+   * This variable describes table id.
+   */
+  private static final String TABLE_ID = "events";
+
+  /**
+   * This variable describes data schema.
+   */
+  private static final String DATA_SCHEMA = "user_event";
+
+/*
+  TO CHECK ERROR HANDLING USE THE TABLE OF INVALID USER EVENTS:
+  TABLE_ID = "events_some_invalid"
+ */
+
+  private ImportUserEventsBigQuery() {
+  }
+
+  /**
+   * Get user event service client.
+   *
+   * @return UserEventServiceClient.
+   * @throws IOException if endpoint is incorrect.
+   */
+  private static UserEventServiceClient getUserEventsServiceClient()
+      throws IOException {
+    return UserEventServiceClient.create();
+  }
+
+  /**
+   * Get import user events from big query request.
+   *
+   * @return ImportUserEventsRequest.
+   */
+  public static ImportUserEventsRequest getImportEventsBigQueryRequest() {
+    // TO CHECK ERROR HANDLING PASTE THE INVALID CATALOG NAME HERE:
+    // DEFAULT_CATALOG = "invalid_catalog_name"
+
+    BigQuerySource bigQuerySource = BigQuerySource.newBuilder()
+        .setProjectId(PROJECT_ID)
+        .setDatasetId(DATASET_ID)
+        .setTableId(TABLE_ID)
+        .setDataSchema(DATA_SCHEMA)
+        .build();
+
+    UserEventInputConfig inputConfig = UserEventInputConfig.newBuilder()
+        .setBigQuerySource(bigQuerySource)
+        .build();
+
+    ImportUserEventsRequest importRequest = ImportUserEventsRequest.newBuilder()
+        .setParent(DEFAULT_CATALOG)
+        .setInputConfig(inputConfig)
+        .build();
+
+    System.out.printf("Import user events from BigQuery source request: %s%n",
+        importRequest);
+
+    return importRequest;
+  }
+
+  /**
+   * Call the Retail API to import user events.
+   *
+   * @throws IOException          from the called method.
+   * @throws InterruptedException when a thread is waiting, sleeping, or
+   *                              otherwise occupied, and the thread is
+   *                              interrupted, either before or during the
+   *                              activity.
+   */
+  public static void importUserEventsFromBigQuery()
+      throws IOException, InterruptedException {
+    ImportUserEventsRequest importBigQueryRequest =
+        getImportEventsBigQueryRequest();
+
+    UserEventServiceClient serviceClient = getUserEventsServiceClient();
+
+    String operationName = serviceClient
+        .importUserEventsCallable()
+        .call(importBigQueryRequest)
+        .getName();
+
+    System.out.printf("OperationName = %s\n", operationName);
+
+    OperationsClient operationsClient = serviceClient.getOperationsClient();
+
+    Operation operation = operationsClient.getOperation(operationName);
+
+    while (!operation.getDone()) {
+      // Polling operation delay until the import task is done.
+      final int awaitDuration = 30;
+
+      getUserEventsServiceClient().awaitTermination(awaitDuration,
```

Comment:
May just do serviceClient.awaitTermination.

Not sure if awaitTermination is the right approach though for tracking long running operations. Probably Thread.sleep is easier.

Reply:
done